pkg/controller/batchrelease/workloads/commons.go

```diff
@@ -243,3 +243,59 @@ func getPodOrdinal(pod *corev1.Pod) int {
 	ord, _ := strconv.Atoi(pod.Name[strings.LastIndex(pod.Name, "-")+1:])
 	return ord
 }
+
+func globalFailureThreshold(global, maxUnavailable *intstr.IntOrString, replicas int32) int32 {
+	globalThreshold := 0
+	if global != nil {
+		globalThreshold, _ = intstr.GetScaledValueFromIntOrPercent(global, int(replicas), true)
+	} else if maxUnavailable != nil {
+		globalThreshold, _ = intstr.GetScaledValueFromIntOrPercent(maxUnavailable, int(replicas), true)
+	}
+	return int32(integer.IntMax(0, globalThreshold))
```

Comment:
interger.IntMin(globalThreshold, workload.maxUnavailable) ？ I think it should not be greater than workload.maxUnavailable.

Reply:
If the failed pods equal maxUnavailable, whether the workload controller will not continue publish?

Reply:
The maxUnavailable calculation workload controller is complicated, it's hard to calculate it for different workload.